spec/features/professors/professors_profiles_spec.rb

```diff
@@ -1,37 +1,39 @@
-require 'spec_helper'
+require 'rails_helper'
 
 describe 'Professors:Profiles', type: :feature do
   context 'when update a professor' do
     let(:professor) { create(:professor) }
 
     before do
       login_as(professor, scope: :professor)
+      visit edit_professor_registration_path
     end
 
-    it 'updates with success when the data are valid' do
-      visit edit_professor_registration_path
+    it 'updates with success when the data are valid', js: true do
+      new_email = 'email@email.com'
 
-      fill_in Professor.human_attribute_name(:email), with: 'email@email.com'
-      fill_in I18n.t('simple_form.labels.defaults.current_password'), with: professor.password
+      fill_in 'professor_email', with: new_email
+      attach_file 'professor_profile_image', FileSpecHelper.image.path
+      fill_in 'professor_current_password', with: professor.password
 
-      submit_form
+      find('input[name="commit"]').click
 
       expect(page).to have_current_path edit_professor_registration_path
       expect(page).to have_selector('div.alert.alert-info',
                                     text: I18n.t('devise.registrations.updated'))
+
+      within('a.nav-link') do
+        expect(page).to have_content(new_email)
+      end
     end
 
-    it 'does not update with invalid date, ao show a error message' do
-      visit edit_professor_registration_path
+    it 'does not update with invalid date', js: true do
+      fill_in 'professor_email', with: 'email'
+      fill_in 'professor_current_password', with: professor.password
 
-      fill_in Professor.human_attribute_name(:email), with: 'email'
-      fill_in Professor.human_attribute_name(:password), with: 'abc123'
-      fill_in Professor.human_attribute_name(:password_confirmation), with: 'abc123'
-      fill_in I18n.t('simple_form.labels.defaults.current_password'), with: professor.password
+      attach_file 'professor_profile_image', FileSpecHelper.pdf.path
 
-      expect do
-        submit_form
-      end.to change(Professor, :count).by(0)
+      find('input[name="commit"]').click
```

Comment:
Can be submit_form('input[name="commit"]')

Reply:
fixed